src/app/app.cpp

```diff
@@ -10,7 +12,8 @@ static Context make_context() {
   auto logger = Logger("shar.log"); // TODO: make configurable
   auto config = [&]{
     try {
-      return Config::parse_from_file("config.json");
+      std::fstream config_file{"config.json"};
```

Comment:
you can check if the file exists without exception cathing (https://en.cppreference.com/w/cpp/filesystem/exists)

Reply:
`catch (const std::exception&)` here is not only for "file not found" error. It also catches:
1. Permission error (file exist, but we have no permissions to open\read it).
2. Invalid json error (file exist, but contain invalid json)
So `fs::exists` will not replace this catch block.